docs/discovery/provisioning.md

```diff
@@ -0,0 +1,456 @@
+```
+
+## Handle Provisioning Requests
+
+The Agents SDK provides a registration method for the agent to implement an interface that is called when certain provisioning events happen in Amplify Central. Once registered the Agents SDK will watch for these events and trigger the appropriate method for the agent o handle that provisioning request.
+
+### Creating and Registering Provisioner
+
+This is the provisioning interface that the agent is expected to implement.
+
+```go
+// Provisioning - interface to be implemented by agents for access provisioning
+type Provisioning interface {
+  ApplicationRequestProvision(ApplicationRequest) RequestStatus
+  ApplicationRequestDeprovision(ApplicationRequest) RequestStatus
+  AccessRequestProvision(AccessRequest) RequestStatus
+  AccessRequestDeprovision(AccessRequest) RequestStatus
+  CredentialProvision(CredentialRequest) (RequestStatus, Credential)
+  CredentialDeprovision(CredentialRequest) RequestStatus
+}
+```
+
+Once implemented the agent may register the structure that implements it with the following call.
+
+```go
+import "github.com/Axway/agent-sdk/pkg/agent"
+
+agent.RegisterProvisioner(provisioningImplementation)
+```
+
+#### RequestStatus return structure
+
+All Provisioning request methods will expect a RequestStatus object to be returned after handing the request.  To create this object the Agent SDK provides a builder with the following methods.
+
+```go
+// RequestStatusBuilder - builder to create new request Status
+type RequestStatusBuilder interface {
+  // SetMessage - set the request Status message
+  SetMessage(message string) RequestStatusBuilder
+  // SetProperties - set the properties of the RequestStatus
+  SetProperties(map[string]string) RequestStatusBuilder
+  // AddProperty - add a new property on the RequestStatus
+  AddProperty(key string, value string) RequestStatusBuilder
+  // Success - set the status as success
+  Success() RequestStatus
+  // Failed - set the status as failed
+  Failed() RequestStatus
+}
+```
+
+The properties that are added to the request status are persisted back to the corresponding resource and will be available for subsequent provisioning/deprovisioning requests for that resource.
+
+Example of a successful provisioning request that adds one property to the status.
+
+```go
+return provisioning.NewRequestStatusBuilder().AddProperty("ApplicationID", "apiid").Success()
+```
+
+Example of a failed provisioning request that adds a failure message.
+
+```go
+return provisioning.NewRequestStatusBuilder().SetMessage("failed for reason x").Failed()
+```
+
+### Application Request Provision and Deprovision
+
+The Application request flow begins with the creation of a ManagedApplication resource in the environment that the agent is managing. The agent SDK will receive an event that the resource was created then check that the resource is in a Pending state before handing it off to the agent. Below is an event diagram to show the process taken.
```

Comment:
add comma
The agent SDK will receive an event that the resource was created, then check that the resource

Reply:
...resource was created, check ... state, before handing...